### DISCUSSION TITLE
Misleading Notfound exception

### DISCUSSION THREAD

**Discussion by phyroslam** [spatie/laravel-model-states#191] [Bugs]
https://github.com/spatie/laravel-model-states/blob/7f8c99024fad7e8ed2ed3453c900c972778ebd6f/src/State.php#L143

This line throws a notFound exception when it should throw a notAllowed one

**Comment by brendt** [comment]
Feel free to submit a PR to fix it :)

**Reply by kosarinin** [reply]
The code is correct. `StateConfig` contains all allowed transitions and since there are no allowed transitions for provided states the `notFound` exception will be thrown. `notAllowed` exception should be thrown only when `Transition::canTransition` leads to false.
